File: Mobile/Winofy/Winofy.Connection/RegisterResult.cs
using System;
using Newtonsoft.Json;

namespace Winofy.Connection
{
    public enum RegisterMessage
    {
        Unknown,
        Ok,
        InvalidRequest,
        UserExists,
        IncorrectUsernameFormat,
        IncorrectPasswordFormat,
    }

    public class RegisterResult
    {
        [JsonProperty("success")]
        public bool Success { get; set; }

        [JsonProperty("messages")]
        public RegisterMessage[] Messages { get; set; }
    }
}

File: Web/winofy-api/results/RegisterResult.go
package results

const (
	RegisterUnknown = "Unknown"
	RegisterOk = "Ok"
	RegisterInvalidRequest = "InvalidRequest"
	RegisterUserExists = "UserExists"
	RegisterIncorrectUsernameFormat = "IncorrectUsernameFormat"
	RegisterIncorrectPasswordFormat = "IncorrectPasswordFormat"
)

type RegisterResult struct {
	Success bool `json:"success"`
	Messages []string `json:"messages"`
}
